Source/OCMock/OCMMacroState.m

```diff
@@ -90,8 +98,7 @@ - (id)initWithRecorder:(OCMRecorder *)aRecorder
 - (void)dealloc
 {
     [recorder release];
-    if(globalState == self)
-        globalState = nil;
+    NSAssert([NSThread currentThread].threadDictionary[OCMGlobalStateKey] != self, @"Unexpected dealloc while set as the global state");
     [super dealloc];
```

Comment:
Why introduce an assert here? And shouldn't the object for `OCMGlobalStateKey` be removed from the thread dictionaory at this point?